python/all-your-base/all_your_base.py

```diff
@@ -0,0 +1,25 @@
+def rebase(input_base, digits, output_base):
+    if output_base <= 1:
+        raise ValueError("output base must be greater than 1")
+    if input_base <= 1:
+        raise ValueError("input base must be greater than 1")
+    if len(list(filter(lambda x: x != 0, digits))) == 0:
+        return []
+
+    # Build up the value in base 10 first to convert into new base
+    base_10_val = 0
+    for i in range(len(digits)):
+        if digits[i] < 0:
+            raise ValueError("digits must all be positive or zero")
+        if digits[i] >= input_base:
+            raise ValueError("digits must be less than the input base")
+        base_10_val += digits[i] * (input_base ** (len(digits) - i - 1))
+
+    new_digits = []
+    while base_10_val > 0:
+        digit = base_10_val % output_base
+        base_10_val = int(base_10_val/output_base)
+        new_digits.insert(0, digit)

```

Comment:
Instead of using `.insert()` here you could `.append()` instead and then reverse the list at the end

```python
new_digits = []
while base_10 > 0:
    # ...
    new_digits.append(digit)

return list(reversed(new_digits))
```

Reply:
Is that a low.zero cost operation in Python?

Reply:
I think it's a low/zero cost operation in any language. Unless I'm mistake inserting at `0` requires shifting the elements of the list and you're doing that on each iteration. Reversing it just requires requires following the pointers backwards...